### PR TITLE
[WIP] Add lowest and highest dependencies tests and fix errors

### PR DESCRIPTION
**Pull request by tgalopin**
As proposed in puli/issues#110, it's a good idea to test the lowest and highest dependecies as well as the current ones.

### CODE REVIEW

**Comment by tgalopin** [comment]
Problem with TestUtil as it didn't exist before.

File: .travis.yml [hunk 1]
```diff
@@ -1,25 +1,30 @@
...
-install: composer install -n
+matrix:
+  include:
+    - php: 5.3
+    - php: 5.4
+    - php: 5.5
+    - php: 5.6
+    - php: 5.6
+    - php: 5.6
+      env: COMPOSER_FLAGS='--prefer-lowest --prefer-stable'
```
Comment (xabbuh) [inline .travis.yml]: What about testing the lowest dependency version with the lowest supported PHP version (5.3 here) instead?

Comment (tgalopin) [inline .travis.yml]: We can do both, but I think we should keep the PHP 5.6-lowest. What do you think @stof @webmozart ? Is it recomended?

Comment (webmozart) [inline .travis.yml]: Not sure. Can you test that on your local machine and whether there is any difference to the 5.6 build?

Comment (tgalopin) [inline .travis.yml]: I don't get what you mean :/ .

I think adding the PHP 5.3-lowest build is interesting to check if the dependencies do work with PHP5.3 but I'm not sure it's very useful as we already have the classic PHP5.3 and the PHP5.6-lowest. It just adds a build for not much value.

**Comment by tgalopin** [comment]
This needs more work: https://travis-ci.org/puli/composer-plugin/jobs/76762729

**Comment by webmozart** [comment]
This must be something with the composer-plugin-api

**Comment by tgalopin** [comment]
Tests are  passing \o/.

I think we should wait to have a stable travis.yml to merge though.

**Pull request merged**